sectionproperties/pre/sections.py

```diff
@@ -1,259 +1,573 @@
+        :param align_to: Another Geometry to align to or a tuple representing an
+        *(x,y)* coordinate point.
+        :type align_to: sectionproperties.pre.sections.Geometry
 
-        :param float angle: Angle (degrees) by which to rotate the section. A positive angle leads
-            to a counter-clockwise rotation.
-        :param rot_point: Point *(x, y)* about which to rotate the section
-        :type rot_point: list[float, float]
+        :param inner: Default False. If True, align the left-most point of this
+        object to the left-most point of 'align_to'. 
+        :type inner: bool
 
-        The following example rotates a 200UB25 section clockwise by 30 degrees::
-
-            import sectionproperties.pre.sections as sections
-
-            geometry = sections.ISection(d=203, b=133, t_f=7.8, t_w=5.8, r=8.9, n_r=8)
-            geometry.rotate_section(angle=-30)
+        :return: Geometry object translated to alignment location
+        :rtype: :class:`sections.pre.sections.Geometry`
         """
+        # Mappings are indexes in the list of bbox extents of both
+        # self and 'align_to'
+        align_self_map = {
+            "left": 0,
+            "right": 1,
+            "bottom": 2,
+            "top": 3,
+        }
+
+        other_as_geom_map = {
+            "left": 1,
+            "right": 0,
+            "bottom": 3,
+            "top": 2,
+        }
+
+        other_as_point_map = {
+            "left": 0,
+            "right": 0,
+            "bottom": 1,
+            "top": 1,
+        }
+
+        self_align_idx = align_self_map[on]
+        align_to_geometry = isinstance(other, Geometry)
+        if align_to_geometry:
+            align_to_idx = other_as_geom_map[on]
+            if inner:
+                align_to_idx = align_self_map[on]
+        else:
+            align_to_idx = other_as_point_map[on]
 
-        # convert angle to radians
-        rot_phi = angle * np.pi / 180
-
-        def get_r(pt1, pt2):
-            """Returns the distance between two points."""
+        self_extents = self.calculate_extents()
+        self_align_coord = self_extents[self_align_idx]
+        if align_to_geometry:
+            other_extents = other.calculate_extents()
+        else:
+            other_extents = other
+        align_to_coord = other_extents[align_to_idx]
 
-            return ((pt1[0] - pt2[0]) ** 2 + (pt1[1] - pt2[1]) ** 2) ** 0.5
+        offset = align_to_coord - self_align_coord
 
-        def get_phi(pt1, pt2):
-            """Returns the angle between two points."""
+        arg = "x_offset"
+        if on in ["top", "bottom"]:
+            arg = "y_offset"
+        kwargs = {arg: offset}
+        new_geom = self.shift_section(**kwargs)
+        return new_geom
 
-            return np.arctan2(pt1[1] - pt2[1], pt1[0] - pt2[0])
+    def align_center(self, align_to: Optional[Geometry] = None):
+        """
+        Returns a new Geometry object, tranlsated in both x and y, so that the 
```

Comment:
small spelling mistake 👍 
```suggestion
        Returns a new Geometry object, translated in both x and y, so that the
```

Reply:
Fixed!